sys/parsoid.js

```diff
@@ -318,7 +318,8 @@ class ParsoidService {
                             // This revision is actually a redirect. Pass redirect target
                             // to caller, and let it rewrite the location header.
                             resp.status = 302;
-                            resp.headers.location = encodeURIComponent(redirectTarget);
+                            // Filter will take care of uri-encoding
+                            resp.headers.location = redirectTarget;
```

Comment:
A that's possible. But that would mean doing encode-decode-encode cycle instead of just one `encode`